libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -346,8 +346,8 @@ def GetOrCreateAnalysisVm(self,
 
   def ListInstanceByLabels(self,
                            labels_filter: Dict[str, str],
-                           filter_union: Optional[bool] = True
-                          )-> Dict[str, 'GoogleComputeInstance']:
+                           filter_union: bool = True
```

Comment:
in some places in the code you use Optional typing and in some other places you remove it, why is this not consistent?

Reply:
See #171 for context

Reply:
I am just thinking that this will be confusing for contributors.

I did a quick test and adding --no-strict-optional [here](https://github.com/google/cloud-forensics-utils/blob/50396979a6e3e330fedb186a4b5942ce9dc0cff3/.github/workflows/pylint_and_mypy.yml#L28) solved the problem. Or you thought about this already and it won't work for other reasons?

That way we can have Optional everywhere without ignore comments.

Reply:
Ah great catch! Didn’t see this option in mypy. I will update PR #186 which was addressing this. I’ll then rebase this branch based on that.

Reply:
Thanks @sa3eed3ed, I updated #186. Let's get that one merged in first.